pkg/exporter/syslogexporter/exporter_test.go

```diff
@@ -24,3 +76,92 @@ func TestInitExporter(t *testing.T) {
 		Format:   "rfc5424"}, createExporterCreateSettings())
 	assert.NoError(t, err)
 }
+
+func prepareExporterTest(t *testing.T, cfg *Config, userPort ...testPort) *exporterTest {
+	// Start a test syslog server
+	var port string
+	var addr net.TCPAddr
+	addr.IP = net.IP{127, 0, 0, 1}
+	addr.Port = 0
+	testServer, err := net.ListenTCP("tcp", &addr)
+	if err != nil {
+		t.Fatal("failed to start test syslog server:", err)
+	}
+	t.Cleanup(func() { testServer.Close() })
+	hostPort := testServer.Addr().String()
+	cfg.Endpoint, port, err = net.SplitHostPort(hostPort)
+	if err != nil {
+		t.Fatalf("could not parse port: %s", err)
+	}
+	if userPort[0].isIncorrect {
+		port = "112"
+	}
```

Comment:
Why do we need it? Why `112`?

Reply:
ok, I understand, but I don't really like passing one boolean as list of structures :thinking:

Reply:
I would rather have a two functions:
- one to get server
- one to build exporter

We can left `func prepareExporterTest(t *testing.T, cfg *Config)` as using the above functions, but for testing incorrect port I would go with the following flow (two steps in test case):

- create server
- build exporter with invalid port

---

Alternatively we can check if `cfg.Port` exists and if yes, we do not override it with server port, but it can be a little magical. We should include some comments in that case

Reply:
Used the first method to split the workflow in two steps.